src/linux/listen.rs

```diff
@@ -166,18 +153,14 @@ unsafe extern "C" fn record_callback(_: *mut i8, raw_data: *mut xrecord::XRecord
     if let Some(event_type) = option_type {
         let name = match event_type {
             EventType::KeyPress(_) => {
-                if let Ok(mut keyboard) = KeyboardState::new() {
-                    keyboard.name_from_code(&mut xdatum.event)
-                } else {
-                    None
-                }
+                KeyboardState::new().map(|mut kboard| kboard.name_from_code(&xdatum.event))
```

Comment:
Neat. Is it idiotmatic to go through a map and flatten ?

Looks wasteful and/or less readable to me, I guess it can be optimized away, no ?

Reply:
flatten was stabilized rather recently, in rust `1.40`, so it hasn't had a long time to make it's way into crates, but it is idiomatic and map is unquestionably idiomatic. I don't actually know that it is optimized out, it just seems like something rust would do.

Reply:
Fine by me.